yrs/src/block.rs

```diff
@@ -51,6 +163,43 @@ pub enum ItemContent {
     Type(types::Inner),
 }
 
+impl ItemContent {
+    pub(crate) fn splice(&mut self, offset: usize) -> Option<ItemContent> {
+        match self {
+            ItemContent::Any(value) => {
+                todo!()
+            },
+            ItemContent::String(string) => {
+                let (left, right) = string.split_at(offset);
+                let mut left = left.to_string();
+                let mut right = right.to_string();
+
+                //TODO: do we need that in Rust?
```

Comment:
The approach in Yjs is that it is not allowed to split surrogate pairs. If you want to change an emoji from :smile:  to :frowning:, then you must replace the whole emoji even if they start with the same character. Fortunately, this is what editors in Yjs already do. If we ever run into issues, we should prevent splitting surrogate pairs with an error message.